modules/react/layout/lib/utils/text.ts

```diff
@@ -1,57 +1,122 @@
 import {Property} from 'csstype';
 
-/** style props to for text properties */
+import {buildStyleFns, buildStylePropFn, StyleFnConfig} from './buildStyleFns';
+import {SystemPropValues} from './systemProps';
+
+/** style props to for CSS font and text properties */
 export type TextStyleProps = {
-  /** sets `line-height` property */
+  /**
+   * - sets [CSS font-family property](https://developer.mozilla.org/en-US/docs/Web/CSS/font-family)
+   * - system tokens: `font` */
+  fontFamily?: Property.FontFamily | SystemPropValues['font'];
+  /**
+   * - sets [CSS font-size property](https://developer.mozilla.org/en-US/docs/Web/CSS/font-size)
+   * - system tokens: `fontSize` */
+  fontSize?: Property.FontSize | SystemPropValues['fontSize'];
+  /** sets [CSS font-style property](https://developer.mozilla.org/en-US/docs/Web/CSS/font-style) */
+  fontStyle?: Property.FontStyle;
+  /**
+   * - sets [CSS font-weight property](https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight)
+   * - system tokens: `fontWeight` */
+  fontWeight?: Property.FontWeight | SystemPropValues['fontWeight'];
+  /** sets [CSS line-height property](https://developer.mozilla.org/en-US/docs/Web/CSS/line-height) */
   lineHeight?: Property.LineHeight;
-  /** sets `letter-spacing` property */
+  /** sets [CSS letter-spacing property](https://developer.mozilla.org/en-US/docs/Web/CSS/letter-spacing) */
   letterSpacing?: Property.LetterSpacing;
-  /** sets `text-align` property */
+  /** sets [CSS text-align property](https://developer.mozilla.org/en-US/docs/Web/CSS/text-align) */
   textAlign?: Property.TextAlign;
-  /** sets `text-decoration` property */
+  /** sets [CSS text-decoration property](https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration) */
   textDecoration?: Property.TextDecoration;
-  /** sets `text-overflow` property */
+  /** sets [CSS text-overflow property](https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow) */
   textOverflow?: Property.TextOverflow;
-  /** sets `text-transform` property */
+  /** sets [CSS text-transform property](https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform) */
   textTransform?: Property.TextTransform;
-  /** sets `text-shadow` property */
+  /** sets [CSS text-shadow property](https://developer.mozilla.org/en-US/docs/Web/CSS/text-shadow) */
   textShadow?: Property.TextShadow;
-  /** sets `white-space` property */
+  /** sets [CSS white-space property](https://developer.mozilla.org/en-US/docs/Web/CSS/white-space) */
   whiteSpace?: Property.WhiteSpace;
-  /** sets `word-break` property */
+  /** sets [CSS word-break property](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break) */
   wordBreak?: Property.WordBreak;
 };
 
-const textProps = [
-  'lineHeight',
-  'letterSpacing',
-  'textAlign',
-  'textDecoration',
-  'textOverflow',
-  'textTransform',
-  'textShadow',
-  'whiteSpace',
-  'wordBreak',
+export const textStyleFnConfigs: StyleFnConfig[] = [
```

Comment:
Originally, we had separate `text` and `font` style prop functions. That made sense because of the types and how values were updated. But with this refactor, it's really simple to combine them.